staging/src/k8s.io/client-go/apiserverupdate/update_test.go

```diff
@@ -52,6 +77,70 @@ func TestSetAPIServerConfig(t *testing.T) {
 	assert.Equal(t, "", ss2.Addresses[0].Hostname)
 }
 
+func setAndReadAPIServerConfig(wg *sync.WaitGroup, epMap map[string]v1.EndpointSubset) {
+	wg.Add(1)
+	SetAPIServerConfig(epMap)
+	readEPMap := GetAPIServerConfig()
+	for sg, ss := range readEPMap {
+		time.Sleep(10 * time.Millisecond)
+		klog.V(6).Infof("Make sure read server group %s and endpoints %v", sg, ss)
+	}
+	wg.Done()
+}
+
+func TestConcurrentReadWriteAPIServerConfig(t *testing.T) {
+	epMap1 := make(map[string]v1.EndpointSubset)
+	epMap2 := make(map[string]v1.EndpointSubset)
+
+	epMap1[serviceGroupId1] = v1.EndpointSubset{
+		Addresses:      []v1.EndpointAddress{{IP: masterIP1}},
+		ServiceGroupId: serviceGroupId1,
+	}
+
+	epMap2[serviceGroupId1] = v1.EndpointSubset{
+		Addresses:      []v1.EndpointAddress{{IP: masterIP1}},
+		ServiceGroupId: serviceGroupId1,
+	}
+
+	epMap2[serviceGroupId2] = v1.EndpointSubset{
+		Addresses:      []v1.EndpointAddress{{IP: masterIP2}},
+		ServiceGroupId: serviceGroupId2,
+	}
+
+	var wg sync.WaitGroup
+	for j := 0; j < 10; j++ {
+		for i := 0; i < 5000; i++ {
+			go setAndReadAPIServerConfig(&wg, epMap1)
+			go setAndReadAPIServerConfig(&wg, epMap2)
+		}
+		wg.Wait()
+	}
+
+	// final test
+	readEPMap := GetAPIServerConfig()
+	assert.NotNil(t, readEPMap)
+	assert.True(t, len(readEPMap) <= 2)
```

Comment:
seems to me that we also need to check assert.True(t, len(readEPMap) >= 1)

Reply:
line 123 does the same thing.